### PR TITLE
Remove `DescendantRebaser` and cleanup `MutableRepo::rebase_descendants_*` functions

### PR DESCRIPTION
**Pull request by bnjmnt4n**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: lib/src/repo.rs [hunk 1]
```diff
@@ -1274,16 +1274,6 @@ impl MutableRepo {
     // TODO(ilyagr): Either document that this also moves bookmarks (rename the
     // function and the related functions?) or change things so that this only
     // rebases descendants.
```
Comment (yuja) [inline lib/src/repo.rs:1276]: nit: orphaned inline comment?

File: lib/src/repo.rs [hunk 2]
```diff
@@ -1275,19 +1275,28 @@ impl MutableRepo {
...
+    ///
+    /// If `options.empty` is the default ([`EmptyBehaviour::Keep`]), all
+    /// rebased descendant commits will be preserved even if they were
+    /// emptied following the rebase operation. A map of newly rebased
+    /// commit ID to original commit ID will be returned.
     ///
-    /// If the `options.empty` is the default, this function will only
-    /// rebase commits, and the return value is what you'd expect it to be.
+    /// Otherwise, this function may rebase some commits and abandon others,
+    /// based on the given [`EmptyBehaviour`]. The behavior is such that only
```
Comment (yuja) [inline lib/src/repo.rs:1289]: nit `EmptyBehavior` isn't in scope, so this link syntax doesn't work.

File: lib/src/repo.rs [hunk 3]
```diff
@@ -1312,13 +1321,31 @@ impl MutableRepo {
...
 
+    /// Rebase descendants of the rewritten commits.
+    ///
+    /// The descendants of the commits registered in `self.parent_mappings` will
+    /// be recursively rebased onto the new version of their parents.
+    /// Returns the number of rebased descendants.
+    ///
+    /// This function is similar to [`MutableRepo::rebase_descendants`], but
+    /// allows for rebase behavior to be customized via [`RebaseOptions`].
     pub fn rebase_descendants_with_options(
```
Comment (yuja) [inline lib/src/repo.rs:1349]: fwiw, we have no callers of `rebase_descendants_with_options()`, and one for `rebase_descendants_return_map()` (except for tests.)

Maybe we can remove these and migrate all callers to `rebase_descendants()` and `rebase_descendants_with_options()` (`_return_map`)?

(no action needed within this PR)

**Review by yuja — APPROVED** [review]
Nice.

File: lib/src/repo.rs (hunk 3 above)
Comment (bnjmnt4n) [reply lib/src/repo.rs:1345]: I guess those are convenience functions for the default behaviour, and perhaps they should be kept in the lib?

Comment (yuja) [reply lib/src/repo.rs:1345]: My feeling is that they're too many. The simplest `rebase_descendants() -> usize` shouldn't be removed, but there aren't many callers of the other variants.

**Review by martinvonz — APPROVED** [review]
Thanks! I agree with Yuya that we should remove the infrequently used versions of the functions.

**Pull request merged**